ydb/library/yql/providers/common/pushdown/collection.cpp

```diff
@@ -355,11 +351,11 @@ bool CheckExpressionNodeForPushdown(const TExprBase& node, const TExprNode* lamb
         return true;
     } else if (settings.IsEnabled(TSettings::EFeatureFlag::ParameterExpression) && node.Maybe<TCoParameter>()) {
         return true;
-    } else if (IsSupportedArithmeticalExpression(node, settings)) {
-        TCoBinaryArithmetic op = node.Cast<TCoBinaryArithmetic>();
-        return CheckExpressionNodeForPushdown(op.Left(), lambdaArg, settings) && CheckExpressionNodeForPushdown(op.Right(), lambdaArg, settings);
+    } else if (const auto op = node.Maybe<TCoUnaryArithmetic>(); op && settings.IsEnabled(TSettings::EFeatureFlag::ArithmeticalExpressions)) {
```

Comment:
This function is also called [here](https://github.com/ydb-platform/ydb/blob/main/ydb/library/yql/providers/generic/provider/yql_generic_physical_opt.cpp#L141C18-L141C18)
If you don't want to support a new expression [here](https://github.com/ydb-platform/ydb/blob/main/ydb/library/yql/providers/generic/provider/yql_generic_predicate_pushdown.h#L12) and futher throught the stack of generic provider, it's better to disable it with another flag in settings for generic provider and enable in OLAP

Reply:
And where is tests for generic provider?

Reply:
`ydb/core/kqp/ut/federated_query/generic`
`ydb/core/kqp/ut/federated_query/generic_ut`